### DISCUSSION TITLE
Unable to work with collision avoidance.

### DISCUSSION THREAD

**Discussion by Pranav-India** [NVIDIA-AI-IOT/jetbot#416] [Q&A]
Hello ,

I am training to run the collision avoidance code on my jetbot. When I try to run the code there are 2 issues 
1st ; to run some cells from the notebook it takes a lot of time and then the kernel is stuck there is no option but to restart the kernel 
2nd ; if I restart the kernel the camera does not start i.e. i am unable to read from the camera it needs me to enter camera.stop() to again get access to the camera . this is gets stuck and I have to restart the bot.

Also I have an hardware issue the battery dies to quickly it does not even sustain for 2 hours but lights on the power bank shows 3 lights on but still it does not work.

**Comment by santaimpersonator** [comment]
Kernel issue
- In the newer jetpack images, using the swap file and running the examples with Docker containers seems to help with the kernel's not getting stuck.
- Also, running at max power helps, because the CPU isn't throttled.

Camera issue
- I have ran into similar issues with the camera. Unfortunately, in my experience only resetting the board fixes the problem.

Battery issue
- You can use a lower power mode, but that also throttles the CPU.
    - It'll make running examples with the camera slow and anything with a pre-trained model take eons to process.
